### PR TITLE
(rio up remoterepo) supporting provided path to Riofile and Riofile-answers relative to repo base, provide current namespace for dynamic configurations

### PR DESCRIPTION
**Pull request by lucidprogrammer**
Closes rancher/rio#730 Closes rancher/rio##816

adds --push-registry-secret option to rio up, changes StackBuild spec

### CODE REVIEW

**Comment by StrongMonkey** [comment]
@lucidprogrammer Thank you for the contributions! Sorry but we decided we are not merging features into v0.6.0, but it will be in the next release. Can you just run your patch for now? We will merge your PR after v0.6.0 is out.

File: cli/cmd/up/up.go [hunk 1]
```diff
@@ -107,7 +111,10 @@ func (u *Up) loadFileAndAnswer(c *clicontext.CLIContext) (string, map[string]str
 	if err != nil {
 		return "", nil, err
 	}
-
+	if answers == nil {
+		answers = map[string]string{}
+	}
+	answers["namespace"] = c.GetSetNamespace()
```
Comment (StrongMonkey) [inline cli/cmd/up/up.go]: Can we make this all upper case so it aligns with [here](https://github.com/rancher/rio/blob/master/stacks/rio-controller-stack.yaml#L7)
```suggestion
	answers["NAMESPACE"] = c.GetSetNamespace()
```

File: modules/build/controllers/stack/stack.go [hunk 2]
```diff
@@ -92,6 +92,23 @@ func (p populator) populateBuild(stack *riov1.Stack, systemNamespace string, os
...
 	}
+	rioUpArgs := []string{
+		"-n",
+		stack.Namespace,
+		"up",
+		"--name",
+		stack.Name,
+	}
+	if stack.Spec.Build.Riofile != "" {
+
```
Comment (StrongMonkey) [inline modules/build/controllers/stack/stack.go]: Nit: remove empty space

File: modules/build/controllers/stack/stack.go [hunk 3]
```diff
@@ -92,6 +92,23 @@ func (p populator) populateBuild(stack *riov1.Stack, systemNamespace string, os
...
+		rioUpArgs = append(rioUpArgs, "--file", stack.Spec.Build.Riofile)
+
+	}
+	if stack.Spec.Build.RiofileAnswers != "" {
+
```
Comment (StrongMonkey) [inline modules/build/controllers/stack/stack.go]: Same

File: cli/cmd/up/up.go [hunk 4]
```diff
@@ -20,6 +20,7 @@ type Up struct {
 	Branch             string   `desc:"Set branch when pointing stack to git repo" default:"master"`
 	BuildCloneSecret   string   `desc:"Set git clone secret name"`
 	BuildWebhookSecret string   `desc:"Set GitHub webhook secret name"`
+	PushRegistrySecret string   `desc:"Set secret for pushing to custom registry"`
```
Comment (cbron) [inline cli/cmd/up/up.go]: Lets add this to the cli ref docs too: https://github.com/rancher/rio/blob/master/docs/cli-reference.md#up

**Pull request merged**